sdk/identity/identity/samples/AzureIdentityExamples.md

```diff
@@ -908,6 +915,215 @@ class RotatingCertificateCredential implements TokenCredential {
+
+The second difference is here the application is preventing the credential from automatically initiating user interaction. Even though the application authenticates the user before the credential is used, further interaction might still be needed, for instance, in the case that the user's refresh token expires or a specific method require additional consent or authentication.
+
+By setting the option `disableAutomaticAuthentication` to `true` the credential will fail to automatically authenticate calls where user interaction is necessary. Instead, the credential will throw an error `AuthenticationRequiredError`. The following example demonstrates an application handling such an exception to prompt the user to authenticate only after some application logic has completed.
+
+```ts
+try {
+  await client.getSecret("secret-name");
+} catch (e) {
+  if (e.name === "AuthenticationRequiredError") {
+    await credential.authenticate(e.scopes);
+    console.log("Secret", await client.getSecret("secret-name"));
+  } else {
+    throw e;
+  }
+}
+```
+
+### Persisting user authentication data
+
+Quite often, applications desire the ability to be run multiple times without re-authenticating the user on each execution. This requires that data from credentials be persisted outside of the application memory to authenticate silently on subsequent executions. Applications can persist this data using `tokenCachePersistenceOptions` when constructing the credential and persisting the `authenticationRecord` returned from `authenticate`. In `@azure/identity` starting from v2 we need to use the package `@azure/identity-cache-persistence` that provides an extension to the identity package to enable persistent token caching. The package `@azure/identity-cache-persistence` exports an extension object that you must pass as an argument to the top-level useIdentityExtension function from the @azure/identity package. Enable token cache persistence in your program as follows:
+
+```ts
+import { useIdentityExtension } from "@azure/identity";
+import { cachePersistenceExtension } from "@azure/identity-cache-persistence";
+
+useIdentityExtension(cachePersistenceExtension);
+```
+
+After calling `useIdentityExtension`, the persistent token cache extension is registered to the `@azure/identity` package and will be available on all credentials that support persistent token caching (those that have `tokenCachePersistenceOptions` in their constructor options).
+
+#### Persisting the token cache
+
+The credential handles persisting all the data needed to silently authenticate one or many accounts. It manages sensitive data such as refresh tokens and access tokens which must be protected to prevent compromising the accounts related to them. By default, the `@azure/identity` library will protect and cache sensitive token data using available platform data protection.
```

Comment:
Use of the term "by default" here is misleading. `@azure/identity` will _only_ use platform data protection if `@azure/identity-cache-persistence` is configured as described above. Otherwise it will use only in-memory caching, if I understand correctly.

CC @sadasant

Reply:
updated